### PR TITLE
Fix some sporadically failing Search tests

### LINKED ISSUE #15936: [Test failure] Cancels_OnlySlow
```

Error message
Expected: False
But was:  True


Stack trace
   at Azure.Search.Documents.Tests.AsyncEventExtensionsTests.Cancels_OnlySlow() in /Users/runner/work/1/s/sdk/search/Azure.Search.Documents/tests/Batching/AsyncEventExtensionsTests.cs:line 450

```

https://dev.azure.com/azure-sdk/public/_build/results?buildId=572950&view=ms.vss-test-web.build-test-results-tab&runId=14102406&resultId=100472&paneView=debug

### PR DESCRIPTION
**Pull request by heaths**
Fixes #15936 and another issue with document counts being off (appears to be from different replicas sometimes).

### CODE REVIEW

**Review by pakrym — COMMENTED** [review]
.

**Review by pakrym — COMMENTED** [review]
Love fixes like these.
<br/>
<br/>
<br/><br/>





Not.

**Review by pakrym — APPROVED** [review]

**Comment by heaths** [comment]
Agreed. Changing to a basic SKU might be a better long-term approach, but something we should discuss for reasons I mentioned in a commit description. Basically: cost (thought, not much of a factor between free and basic), availability, robustness, and provisioning time.

File: sdk/search/Azure.Search.Documents/tests/Samples/Sample05_IndexingDocuments.cs [hunk 1]
```diff
@@ -121,6 +121,13 @@ public async Task SimpleIndexing()
 
                 await WaitForDocumentCountAsync(searchClient, 1000);
 
+                // When using the free SKU, there may be enough load to prevent
+                // immediately replication to all replicas and we get back the
+                // wrong count. Wait another second before checking again. We
+                // may also upgrade to a basic SKU, but that will take longer
+                // to provision.
+                await DelayAsync(TimeSpan.FromSeconds(1));
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/tests/Samples/Sample05_IndexingDocuments.cs:129]: Thanks for looking at this, but why are we adding a delay after the `WaitForDocumentCountAsync` above?  That also has parameters that will let you stretch the number of times it checks and the amount it waits between checks if you want extra delay.  I believe it does 3 attempts today with a second between each.  We could bump that to two seconds, etc., if it's not been reliable enough in practice.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/tests/Samples/Sample05_IndexingDocuments.cs:129]: The comment states that replicas need time to get updates. This is what @brjohnstmsft recommends. Given more test failures last night with this same problem, I'm moving this to `WaitForDocumentCountAsync` or we're just playing whack-a-mole. If you want more information, let's chat offline.

**Pull request merged**